### PR TITLE
#156: fix issue with wrong namespaces

### PR DESCRIPTION
**Pull request by mmezhensky**
This PR fixed issue with wrong namespaces.

### Description (*)

This PR fixed two issues with namespace:

1. A wrong namespace for the first method parameter in the plugin if the name of the plugin and class are the same. I added the full path for first method parameter

1. I fixed the wrong namespace for the second method parameter in the plugin.

### Fixed Issues (if relevant)

1. magento/magento2-phpstorm-plugin#156: Wrong namespace for method parameters in plugin

### Steps to reproduce (*)

- **for the first issue:**
1. open file Magento\Catalog\Api\ProductRepositoryInterface
2. create a new plugin. >> Set plugin class name is like the same as the name of the file (ProductRepositoryInterface)
3. open new plugin file >> The first parameter must be with the full namespace.

- **for the second issue:**
1. open file Magento\Catalog\Api\ProductRepositoryInterface
2. create a new "before" plugin for save()
3. open new plugin file >> The second parameter must be with the right namespace.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/generator/code/PluginMethodsGenerator.java [hunk 1]
```diff
@@ -202,9 +214,19 @@ protected String getParameterList(Collection<PsiElement> parameters, @NotNull Pl
...
+                    String[] fqnArray = parameterText.split("\\\\");
+                    parameterText = fqnArray[fqnArray.length - 1];
+                }
+                buf.append(parameterText);
             } else {
-                String typeHint = this.getTypeHint(element);
+                Boolean globalType = true;
+                if (i == 0) {
+                    globalType = false;
+                }
```
Comment (coderimus) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/code/PluginMethodsGenerator.java:227]: Not critical, but with ternary operator should look better :)
```java
Boolean globalType = (i == 0) ? false : true;
```

File: src/com/magento/idea/magento2plugin/actions/generation/generator/code/PluginMethodsGenerator.java [hunk 2]
```diff
@@ -257,21 +279,21 @@ protected String getReturnVariables(@NotNull Plugin.PluginType type) {
...
 
         if (typeStrings.size() == 2) {
             PhpType filteredNullType = filterNullCaseInsensitive(filedType);
             if (filteredNullType.getTypes().size() == 1) {
                 if (PhpLanguageFeature.NULLABLES.isSupported(element.getProject())) {
-                    typeString = "?" + this.convertTypeToString(element, filteredNullType.getTypes());
+                    typeString = "?" + this.convertTypeToString(element, filteredNullType.getTypes(), globalType);
                 } else {
-                    typeString = this.convertTypeToString(element, filteredNullType.getTypes());
+                    typeString = this.convertTypeToString(element, filteredNullType.getTypes(), globalType);
```
Comment (coderimus) [inline src/com/magento/idea/magento2plugin/actions/generation/generator/code/PluginMethodsGenerator.java:296]:
```java
typeString = this.convertTypeToString(element, filteredNullType.getTypes(), globalType);
if (PhpLanguageFeature.NULLABLES.isSupported(element.getProject())) {
    typeString = "?" + typeString;
}
```

**Comment by mmezhensky** [comment]
Hi @coderimus!
I added a fix for these changes.
Thanks for your review!

**Comment by coderimus** [comment]
@mmezhensky thanks for changes 👍 

QA passed for all plugin types ✅ 
image

**Comment by VitaliyBoyko** [comment]
@coderimus @mmezhensky Thanks, guys!

**Pull request merged**